### DISCUSSION TITLE
Consistent branch naming not depending on the repo branches

### DISCUSSION THREAD

**Discussion by ns-ggeorgiev** [renovatebot/renovate#24431] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have central configuration for many repos. I recently added support for lets call them feature branches to be scanned from renovate.

Now when a repo is having at least one feature branch all renovate branch names targeting master get changed to include the target branch. When the a feature branch is created all issues get recreated, when the last feature branch is deleted, the issues are recreated again.

That by itself is not a big deal, but we have internal jira integration that uses the branch to identify the jira tickets for particular issue.

I am looking for consistency. How I can configure branchXYZ family to give me consistency of the used branches. Either way:
1. Setting up that the renovate branches always have the target branch in the name
2. Setting up that the *default branch does not have the target branch in the name ever, even if there are other branches in the scan

### What you tried so far.

looked at the code in lib/workers/repository/process/index.ts

```
  if (config.baseBranches!.length > 1) {
    baseBranchConfig.branchPrefix += `${baseBranch}-`;
    baseBranchConfig.hasBaseBranches = true;
  }
```
  
  and if I am right that this is the place where this is determined, there is actually no way I to achieve what I want without a code change.
  
  and likely this will accomplish option 2
  
```
  if (config.baseBranches!.length > 1) {
    if (baseBranch !== baseBranchConfig.defaultBranch) {
      baseBranchConfig.branchPrefix += `${baseBranch}-`;
    }
    baseBranchConfig.hasBaseBranches = true;
  }
```
  
  
  
  
  

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Any change to Renovate's built-in logic for this would result in a lot of people's existing branches/PRs being closed and recreated, so it would be best to make any such change configurable/opt-in, such as baseBranchAlwaysPrefix=true

**Reply by ns-ggeorgiev** [reply]
would it make sense if this code is replaces with some sort of template code that implements the current behavior

say `{{#if config.baseBranches!.length > 1)}}{{{baseBranch}}}-{{/if}` or whatever it has to look like - I am not an expert.

This way to accomplish what I want will simple be to override the default prefix with `{{{baseBranch}}}`

**Reply by rarkins** [reply]
Yes, potentially something like `{{baseBranchesPrefix}}` is added to the branch (like additionalBranchPrefix) and then that has a template you can override
